sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -389,6 +389,50 @@ case class TruncateTableCommand(
 }
 
 /**
+ * A command to repair a table by discovery all the partitions in the directory.
+ *
+ * The syntax of this command is:
+ * {{{
+ *   MSCK REPAIR TABLE table_name;
+ * }}}
+ *
+ * This command is the same as AlterTableRecoverPartitions
+ */
+case class RepairTableCommand(tableName: TableIdentifier) extends RunnableCommand {
+  override def run(spark: SparkSession): Seq[Row] = {
+    val catalog = spark.sessionState.catalog
+    val table = catalog.getTableMetadata(tableName)
+    if (!catalog.tableExists(tableName)) {
+      throw new AnalysisException(s"Table $tableName in MSCK REPAIR TABLE does not exist.")
+    }
+    if (catalog.isTemporaryTable(tableName)) {
+      throw new AnalysisException(
+        s"Operation not allowed: MSCK REPAIR TABLE on temporary tables: $tableName")
+    }
+    if (DDLUtils.isDatasourceTable(table)) {
+      throw new AnalysisException(
+        s"Operation not allowed: MSCK REPAIR TABLE on datasource tables: $tableName")
+    }
+    if (table.tableType != CatalogTableType.EXTERNAL) {
+      throw new AnalysisException(
+        s"Operation not allowed: MSCK REPAIR TABLE only works on external tables: $tableName")
+    }
+    if (DDLUtils.isTablePartitioned(table)) {
```

Comment:
This checking conflicts with the error message reported below. Now, if the table is partitioned, we will report the message: `MSCK REPAIR TABLE only works on partitioned tables`